core/vm/lightclient/v2/lightclient.go

```diff
@@ -118,12 +118,19 @@ func (cs *ConsensusState) ApplyLightBlock(block *types.LightBlock) (bool, error)
 		}
 	}
 
+	valSetChanged := !(bytes.Equal(cs.ValidatorSet.Hash(), block.ValidatorsHash))
+
 	// update consensus state
 	cs.Height = uint64(block.Height)
 	cs.NextValidatorSetHash = block.NextValidatorsHash
 	cs.ValidatorSet = block.ValidatorSet
 
-	return !(bytes.Equal(cs.ValidatorSet.Hash(), block.ValidatorsHash)), nil
+	if !isHertz {
+		// This logic is wrong, fixed in hertz fork.
+		return !(bytes.Equal(cs.ValidatorSet.Hash(), block.ValidatorsHash)), nil
```

Comment:
the `ValidatorSet` was updated in line#126, it will make the change the behavior before Hertz?